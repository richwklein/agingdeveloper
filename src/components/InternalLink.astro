---
/**
 * Component used to render a link internal to the site.
 *
 * @param {Props} props - The properties of the link.
 * @return {Function} an Astro component factory function.
 *
 * @example
 * <InternalLinkProps to="tags">Tags</InternalLink>
 */

interface Props {
  to: string;
  class?: string;
}

const { to, class: className } = Astro.props;
---

<!--prettier-ignore -->
<a data-link="internal" target="_self" href={to} class:list={[className]}><slot /></a>
