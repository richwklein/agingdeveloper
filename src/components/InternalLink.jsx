import React from "react";
import MuiLink from "@mui/material/Link";
import {Link as GatsbyLink} from "gatsby";
import {LinkProps} from "../props";


/**
 * React component used to render a link internal to the site.
 *
 * @param {LinkProps} props - The to link and rest.
 * @return {React.ReactElement} - The react component
 *
 * @example
 * <InternalLinkProps to="tags">Tags</InternalLink>
 * @todo Remove prop spread
 */
const InternalLink = ({to, ...rest}) => (
  <MuiLink data-link="internal" component={GatsbyLink} to={to} {...rest} />
);

InternalLink.propTypes = LinkProps;

export default InternalLink;
