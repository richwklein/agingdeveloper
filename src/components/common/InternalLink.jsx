import React from "react";
import MuiLink from "@mui/material/Link";
import {Link as GatsbyLink} from "gatsby";

const InternalLink = React.forwardRef((props, ref) => (
  <MuiLink data-link="internal" component={GatsbyLink} ref={ref} {...props} />
));

InternalLink.displayName = "InternalLink";

export default InternalLink;
