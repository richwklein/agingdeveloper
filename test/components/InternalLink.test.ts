import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { describe, expect, test } from "vitest";
import InternalLink from "../../src/components/InternalLink.astro";

describe("InternalLink", () => {
  const render = async () => {
    const container = await AstroContainer.create();
    return container.renderToString(InternalLink, {
      slots: { default: "Test Content" },
      props: { to: "https://example.com" },
    });
  };

  test("that it renders correctly.", async () => {
    const result = await render();
    expect(result).toMatchSnapshot();
  });

  test("that it has a data-link of internal.", async () => {
    const result = await render();
    expect(result).toContain('data-link="internal"');
  });

  test("that it has same window target.", async () => {
    const result = await render();
    expect(result).toContain('target="_self"');
  });

  test.todo("passing an additional attribute");
});
