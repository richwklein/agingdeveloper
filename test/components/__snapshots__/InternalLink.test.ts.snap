// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InternalLink > that it renders correctly. 1`] = `"<!DOCTYPE html><!--prettier-ignore --><a data-link="internal" target="_self" href="https://example.com" data-astro-source-file="/Users/richard/Development/Projects/agingdeveloper-astro/src/components/InternalLink.astro" data-astro-source-loc="21:69">Test Content</a>"`;
